sycl/plugins/cuda/pi_cuda.hpp

```diff
@@ -440,6 +444,43 @@ struct _pi_queue {
     }
   }
 
+  template <typename T> void sync_streams(T &&f) {
+    {
+      std::lock_guard<std::mutex> compute_guard(compute_stream_mutex_);
+      unsigned int size = static_cast<unsigned int>(compute_streams_.size());
+      unsigned int start = last_sync_compute_streams_;
+      unsigned int end = num_compute_streams_ < size
+                             ? num_compute_streams_
+                             : compute_stream_idx_.load() % size;
```

Comment:
Consider a case where we have `size == 8` and I sync with the queue when `compute_stream_idx_` is 3. If I am reading this correctly it will sync with streams 0, 1, 2, and 3.

Let's say we now enqueue some more work to that queue and then sync again when `compute_stream_idx_` is 11. This means we've enqueued work to all 8 compute streams, but when we sync this will have `start = 3` and `end = compute_stream_idx_ % size` which is 3, so it will only sync with compute stream 3, when it should be synching with all 8 streams.

Reply:
fixed

Reply:
Indeed! Thank you. 😄